### PR TITLE
fix(provider/kubernetes): Support the renaming of the heptio-authenticator-aws to aws-iam-authenticator

### PR DESCRIPTION
**Pull request by nicholasgcoles**
The `heptio-authenticator-aws` has been renamed to `aws-iam-authenticator`, this means that when trying to set up pipelines / jobs that the following error message occurs:
```
Exception ( Monitor Deploy )
Deploy failed: error: unable to recognize "STDIN": Get https://XYZ.us-east-1.eks.amazonaws.com/api?timeout=32s: getting credentials: exec: exec: "aws-iam-authenticator": executable file not found in $PATH
```
In this pull request I have changed the Dockerfile to pull the latest `aws-iam-authenticator`. My thinking is to create a link from `aws-iam-authenticator` to `heptio-authenticator-aws` for now so that there is no breaking change.

### CODE REVIEW

**Comment by edwinavalos** [comment]
@nicholasgcoles can you rebase with master, and I can get this merged.

File: Dockerfile.slim [hunk 1]
```diff
@@ -16,16 +16,17 @@ RUN wget https://storage.googleapis.com/kubernetes-release/release/stable.txt &&
...
+  mv ./aws-iam-authenticator /usr/local/bin/aws-iam-authenticator && \
+  ln -sf /usr/local/bin/aws-iam-authenticator /usr/local/bin/heptio-authenticator-aws
 
 RUN apk -v --update add py-pip && \
   pip install --upgrade awscli==1.14.5 s3cmd==2.0.1 python-magic && \
   apk -v --purge del py-pip && \
   rm /var/cache/apk/*
 
-ENV PATH "$PATH:/usr/local/bin/heptio-authenticator-aws"
+ENV PATH "$PATH:/usr/local/bin/aws-iam-authenticator"
```
Comment (edwinavalos) [inline Dockerfile.slim]: I think you only need to add /usr/local/bin to the path and can drop the aws-iam-authenticator portion here

Comment (nicholasgcoles) [reply Dockerfile.slim]: @edwinavalos, cool done!

**Comment by costimuraru** [comment]
could we get this also in 1.12.3?

**Comment by jtk54** [comment]
@spinnakerbot cherry-pick 1.12

**Pull request merged**